offloaders/aws-s3/2.5.1/aws-s3.md

```diff
@@ -0,0 +1,551 @@
+
+2. In the navigation pane, click **Users** > **Add user**. 
+
+![](/images/offloaders/aws-s3/create-user-1.png)
+
+
+3. Type your **user name** (not case-sensitive) and select **AWS access type**. 
+   
+![](/images/offloaders/aws-s3/create-user-2.png)
+
+
+1. Click **Next Permissions**.
+
+
+2. On the **Set permissions** page, specify how you want to assign permissions to your user. 
+
+![](/images/offloaders/aws-s3/create-user-3.png)
+
+6. Click **Next Tags**.
+
+7. (Optional) Set tags for your user and click **Next: Review**.
+
+> #### Tip
+>
+> For more information about using tags in IAM, see [tagging IAM users and roles](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_tags.html).
+
+![](/images/offloaders/aws-s3/create-user-4.png)
+
+8. Check all of the choices you made up to this point. When you are ready to proceed, choose **Create user**.
+
+![](/images/offloaders/aws-s3/create-user-5.png)
+
+9. To view the users' access keys (`access key IDs` and `secret access keys`), click **Show** next to each password and access key that you want to see. 
+
+    The `access key IDs` should be the same as the value of `AWS_ACCESS_KEY_ID` and the `secret access keys` should be the same as the value of `AWS_SECRET_ACCESS_KEY` that you configured in [Step 1: configure AWS S3 offloader driver](#step-1-configure-aws-s3-offloader-driver).
+
+    > #### Note
+    > 
+    >  To save the access keys, click **Download .csv** and then save the file to a safe location.
+    This is your **only opportunity** to view or download the secret access keys, and you **must** provide this information to your users before they can use the AWS API. Make sure you have saved the user's new access key ID and secret access key in a safe and secure place. You **do not have access** to the secret keys again after this step.
+
+    ![](/images/offloaders/aws-s3/create-user-6.png)
+
+
+## Step 5: offload data from BookKeeper to AWS S3
+
+Execute the following commands in the repository where you download Pulsar tarball. For example, `~/path/to/apache-pulsar-2.5.1`.
+
+1. Start Pulsar standalone.
+
+    ```
+    ./bin/pulsar standalone -a 127.0.0.1
+    ```
+
+2. To ensure the data generated is not deleted immediately, it is recommended to set the [retention policy](https://pulsar.apache.org/docs/en/next/cookbooks-retention-expiry/#retention-policies), which can be either a **size** limit or a **time** limit. The larger value you set for the retention policy, the longer the data can be retained.
+
+    ```
+    ./bin/pulsar-admin namespaces set-retention public/default --size -10G --time 3d
+    ```
+
+    > #### Tip
+    >
+    > For more information about the `pulsar-admin namespaces set-retention options` command, including flags, descriptions, and default values, see [here](http://pulsar.apache.org/tools/pulsar-admin/2.6.0-SNAPSHOT/#-em-set-retention-em-). 
+
+3. To configure AWS S3 offloader, you can set offloader policy at the namespace level.
+
```

Comment:
hello @Anonymitaet , The action of set-offload-policies is a side-by-side operation when we configure various options in the conf file at the beginning. If we configure these attributes in the conf, we do not need to use set-offload-policies to set here .

Reply:
OK